typescript/node-js/client/__tests__/src/app/page.test.tsx

```diff
@@ -0,0 +1,158 @@
+    expect(screen.getByLabelText("Username")).toBeInTheDocument();
+    expect(screen.getByLabelText("Password")).toBeInTheDocument();
+    expect(screen.getByTestId("page-login-btn")).toBeInTheDocument();
+  });
+
+  test("shows error message for empty username or password and handle close", async () => {
+    jest.useFakeTimers();
+
+    render(<Home />);
+    fireEvent.click(screen.getByTestId("page-login-btn"));
+    checkTextToBeInDocument("Username or password cannot be empty");
+
+    act(() => {
+      jest.advanceTimersByTime(6000);
+    });
+
+    await waitFor(() => {
+      expect(screen.queryByText("Username or password cannot be empty")).toBe(
+        null
+      );
+    });
+  });
+
+  test("Handle keyDown Enter on password field", async () => {
+    render(<Home />);
+    fireEvent.keyDown(screen.getByLabelText("Password"), {
+      key: "Enter",
+      code: "Enter",
+    });
+
+    checkTextToBeInDocument("Username or password cannot be empty");
+  });
+
+  test("Handle keyDown do nothing on password field", async () => {
+    render(<Home />);
+    fireEvent.keyDown(screen.getByLabelText("Password"), {
+      key: "shift",
+      code: "shift",
+    });
+
+    await waitFor(() => {
+      expect(fetch).not.toHaveBeenCalled();
+      expect(mockPush).not.toHaveBeenCalled();
+    });
+  });
+
+  test("handles login successfully", async () => {
+    (fetch as jest.Mock).mockResolvedValueOnce({
+      json: async () => ({ status: 200, message: "Success" }),
+    });
+
+    render(<Home />);
+    fireEvent.change(screen.getByLabelText("Username"), {
+      target: { value: "testuser" },
+    });
+    fireEvent.change(screen.getByLabelText("Password"), {
+      target: { value: "password" },
+    });
+    fireEvent.keyDown(screen.getByLabelText("Password"), {
+      key: "Enter",
+      code: "Enter",
+    });
+
+    await waitFor(() => {
+      expect(fetch).toHaveBeenCalledWith(`${MYNAMASTE_URL}/api/login`, {
+        method: "POST",
+        body: JSON.stringify({ username: "testuser", password: "password" }),
+      });
+      expect(mockPush).toHaveBeenCalledWith("/users");
+    });
+  });
+
+  test("shows error message for failed login", async () => {
+    (fetch as jest.Mock).mockResolvedValueOnce({
+      json: async () => ({ status: 401, message: "Invalid credentials" }),
+    });
+
+    render(<Home />);
+    fireEvent.change(screen.getByLabelText("Username"), {
+      target: { value: "testuser" },
+    });
+    fireEvent.change(screen.getByLabelText("Password"), {
+      target: { value: "password" },
+    });
+    fireEvent.click(screen.getByTestId("page-login-btn"));
+
+    await waitFor(() => {
+      expect(fetch).toHaveBeenCalledWith(`${MYNAMASTE_URL}/api/login`, {
+        method: "POST",
+        body: JSON.stringify({ username: "testuser", password: "password" }),
+      });
+      checkTextToBeInDocument("Invalid credentials");
+    });
+  });
+
+  test("shows error message when error raised", async () => {
```

Comment:
bedanya yg ini, ama test sebelumnya apa ya? secara code sama, tapi check nya beda

Reply:
Updated to be clearer. Yang satunya raise error, yg satunya status fetch yg bukan 200
https://github.com/gl-zkPass/zkpass-sdk/pull/85/commits/8672d70ae6139becfc82e9f74e1aeebdb25b312e